src/client/datascience/variablesView/variableViewNotebookWatcher.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import { inject, injectable } from 'inversify';
+import { Event, EventEmitter } from 'vscode';
+import '../../common/extensions';
+import { IFileSystem } from '../../common/platform/types';
+import { IDisposableRegistry } from '../../common/types';
+import { KernelState, KernelStateEventArgs } from '../notebookExtensibility';
+import { INotebook, INotebookEditor, INotebookEditorProvider, INotebookExtensibility } from '../types';
+import { IVariableViewNotebookWatcher } from './types';
+
+// For any class that is monitoring the variables of the active notebook document, this class will update you
+// when the active notebook changes or if the execution count is updated on the active notebook
+// NOTE: Currently this class is only looking at native notebook documents
+@injectable()
+export class VariableViewNotebookWatcher implements IVariableViewNotebookWatcher {
```

Comment:
Could this be renamed as just a notebook watcher? It's not specific to variable views.

Reply:
Yeah, that makes sense, the whole point is that any UI concerned with what notebook is active and when it updates could use it. I'll change that.